### ISSUE TITLE
Implement checksums properly in dataplane

### ISSUE THREAD

**Issue by shaneutt** [kubernetes-sigs/blixt#25]
## Problem Statement

We currently don't handle checksums properly or at all in several places. The purpose of this task is to get that sorted out.

## Acceptance Criteria

- [ ] find all the places we are currently handling, or NOT handling L3 and L4 checksums and get them consistent
- [ ] use helpers like `bpf_l3_cksum_replace` to properly recalculate checksums

**Comment by astoycos** [comment]
Opening this up as a good issue for newcomers interested in getting involved in the ebpf dataplane bits :)

**Comment by LCaparelli** [comment]
Would like to work on this one o/

**Comment by LCaparelli** [comment]
Yup, the nice folks over there already gave me a hand on the other issue I'm working on to improve the build process, thanks for the heads up @astoycos

**Comment by shaneutt** [comment]
@LCaparelli as I understand it you may not have time for this one going forward so I'm opening it back up, however if you still want to take this one on please just let us know! :vulcan_salute:

**Comment by shaneutt** [comment]
Not stale, we still intend this for the `v0.7.0` milestone:

/remove-lifecycle stale

However, I suspect @LCaparelli is no longer working on this (do correct me if you still wanted this)? So I think we're back to help wanted.

/help

**Comment by maborosii** [comment]
I have some questions about this issue, I found some functions implemented with `bpf_l4_cksum_replace` and `bpf_l3_cksum_replace` like `set_ipv4_dest_port` and `set_ipv4_ip_dst` in tc_ingress module, so you mean that we should keep consistent in other module like tc_egress?

**Comment by shaneutt** [comment]
Consistency for sure, but I also think we just have a lot of hacks in place. I removed the `good-first-issue` on this one some time ago because basically what we need is someone to deep dive in and suss it all out _first_ and actually figure out exactly what is needed, and _then_ make the patch. So just know that we're happy to support you in this one but it's a bit of a deeper dive!

**Comment by jokestax** [comment]
Hey guys,i am happy to work on this task

**Comment by shaneutt** [comment]
@jokestax mentioned wanting to have a go at this one:

/assign @jokestax 

Please do note it's kinda old, and maybe a bit crunchy, so take your time and do reach out here or on Slack if you need any support. Also when in doubt, I would highly recommend looking at what Cilium does as they are usually a good model to follow.

**Comment by jokestax** [comment]
Hey guys,
I've made some progress! The ingress implements the checksum properly. However, I noticed it wasn’t being handled correctly in egress for ICMP and TCP. I’ve made the necessary changes for TCP and raised a draft PR for review: [link](https://github.com/kubernetes-sigs/blixt/pull/325). 

I’d really appreciate it if someone could review the PR and provide guidance so I can proceed with making similar changes for ICMP as well.

**Comment by shaneutt** [comment]
Awesome! Thanks for taking a look at this and putting in a PR, I'll try to review this before the end of the week (but bare with me, just got back from holiday break so I'm likely to be a bit slow).
